### ISSUE TITLE
added .cnf affects image initialization 

### ISSUE THREAD

**Issue by taylorchu** [docker-library/mysql#490]
If mysql `.cnf` is added when container starts, and it has a `[client]` section, incorrect password will be used in the init script.

```
[client]
user=root
password=unprotected
database=xxx_development
```

then init will fail like this:

```
mysql_1                  | MySQL init process in progress...
mysql_1                  | 2018-09-13T02:53:44.017882Z 29 [Note] Access denied for user 'root'@'localhost' (using password: YES)
mysql_1                  | MySQL init process in progress...
mysql_1                  | 2018-09-13T02:53:45.025505Z 30 [Note] Access denied for user 'root'@'localhost' (using password: YES)
mysql_1                  | MySQL init process in progress...
mysql_1                  | 2018-09-13T02:53:46.032175Z 31 [Note] Access denied for user 'root'@'localhost' (using password: YES)
mysql_1                  | MySQL init process in progress...
mysql_1                  | MySQL init process failed.
```

We should probably explicitly use password here:
https://github.com/docker-library/mysql/blob/b39f1e5e4ec82dc8039cecc91dbf34f6c9ae5fb0/5.7/docker-entrypoint.sh#L123

**Comment by ltangvald** [comment]
Maybe best to use --no-defaults for the client commands during init

**Comment by taylorchu** [comment]
@ltangvald yes. will someone fix this today? it seems simple.

**Comment by ltangvald** [comment]
This should be fixed by pr #471 at least, but that may take some time.
As a workaround, maybe you can use a user initiated by MYSQL_USER instead?
If you supply MYSQL_USER, MYSQL_PASSWORD and MYSQL_DATABASE, the container will automatically create a user with that password that has full access to MYSQL_DATABASE
